uber/site_sections/registration.py

```diff
@@ -82,6 +85,18 @@ def index(self, session, message='', page='0', search_text='', uploaded_id='', o
     def form(self, session, message='', return_to='', omit_badge='', **params):
         attendee = session.attendee(params, checkgroups=['interests','requested_depts','assigned_depts'],
                                     bools=['staffing','trusted','international','placeholder','got_merch','can_spam'])
+
+        if 'watchlist_id' in params:
+            watchlist_entry = session.watch_list(params['watchlist_id'])
+            watchlist_entry.disabled = True
+            session.add(watchlist_entry)
+            session.commit() # Need to commit now or the attendee will be automatically set to Deferred
+
+            if attendee.status == DEFERRED_STATUS:
+                attendee.status = NEW_STATUS
+            session.add(attendee)
```

Comment:
Ditto for the ``attendee`` here, which unless I'm mistaken must always be part of the session already if we've entered this if block, since we only enter this if block for existing attendees and never for new ones.